idaes/models/properties/modular_properties/eos/ceos.py

```diff
@@ -447,6 +462,144 @@ def rule_delta_eq(m, p1, p2, p3, i):
+                      EoS_w * B_eq**2 +
+                      EoS_w * B_eq**3)
+                return (_b, _c, _d)
+
+            def second_derivative(b, p1, p2, p3):
+                _b, _, _ = calculate_cubic_coeffs(b, p1, p2, p3)
+                z = b.compress_fact_phase[p3]
+                return 6 * z + 2 * _b
+            b.cubic_second_derivative = Expression(b.params._pe_pairs,
+                                                   b.phase_list,
+                                                   rule=second_derivative)
+
+
+    @staticmethod
+    def build_critical_properties(b, p):
+        if p == 'Vap':
+            return None
+        else:
+            reference_phase = p
+            pobj = b.params.get_phase(reference_phase)
+            cname = pobj.config.equation_of_state_options['type'].name
+            ctype = pobj._cubic_type
+            
+            # Add components for calculation of mixture critical properties
+            def func_a_crit(m, j):
+                cobj = m.params.get_component(j)
+                fw = getattr(m, cname+"_fw")
+                return (EoS_param[ctype]['omegaA']*(
+                            (Cubic.gas_constant(b) *
+                            cobj.temperature_crit)**2/cobj.pressure_crit) *
+                        ((1+fw[j]*(1-sqrt(m.temperature_crit_mix /
+                                          cobj.temperature_crit)))**2))
+            b.add_component('a_crit',
+                            Expression(b.component_list,
+                                        rule=func_a_crit,
+                                        doc='Component a coefficient'))
+            
+            def rule_am_crit(m):
+                try:
+                    rule = m.params.get_phase(reference_phase). \
+                        config.equation_of_state_options["mixing_rule_a"]
+                except KeyError:
+                    rule = MixingRuleA.default
+        
+                a_crit = getattr(m, "a_crit")
+                if rule == MixingRuleA.default:
+                    return rule_am_crit_default(m, cname, a_crit)
+                else:
+                    raise ConfigurationError(
+                        "{} Unrecognized option for Equation of State "
+                        "mixing_rule_a: {}. Must be an instance of MixingRuleA "
+                        "Enum.".format(m.name, rule))
+            b.add_component('am_crit', Expression(rule=rule_am_crit))
+        
+            def rule_bm_crit(m):
+                try:
+                    rule = m.params.get_phase(reference_phase). \
+                        config.equation_of_state_options["mixing_rule_b"]
+                except KeyError:
+                    rule = MixingRuleB.default
+        
+                b = getattr(m, cname+"_b")
+                if rule == MixingRuleB.default:
+                    return sum(m.mole_frac_comp[i] * b[i]
+                                for i in m.component_list)
```

Comment:
For default mixing rules, `bm` isn't a function of temperature, so you could just return the normal `bm`.